### PR TITLE
Indicate submenu item level in admin nav menu editor and customizer nav menu editor

### PR DESCRIPTION
**Pull request by Rcreators**
Trac ticket: https://core.trac.wordpress.org/ticket/60673

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: src/wp-admin/includes/class-walker-nav-menu-edit.php [hunk 1]
```diff
@@ -182,7 +182,7 @@ public function start_el( &$output, $data_object, $depth = 0, $args = null, $cur
 							'<a class="item-edit" id="edit-%s" href="%s" aria-label="%s"><span class="screen-reader-text">%s</span></a>',
 							$item_id,
 							esc_url( $edit_url ),
-							esc_attr__( 'Edit menu item' ),
+							__( 'Edit menu item' ),
```
Comment (joedolson) [inline src/wp-admin/includes/class-walker-nav-menu-edit.php]: This looks like an unintentional change; remove it.

File: src/wp-admin/nav-menus.php [hunk 2]
```diff
@@ -580,10 +580,12 @@
...
 	/* translators: %s: Previous item name. */
 	'outFrom'                 => __( 'Out from under %s' ),
-	/* translators: 1: Item name, 2: Item position, 3: Total number of items. */
-	'menuFocus'               => __( '%1$s. Menu item %2$d of %3$d.' ),
-	/* translators: 1: Item name, 2: Item position, 3: Parent item name. */
-	'subMenuFocus'            => __( '%1$s. Sub item number %2$d under %3$s.' ),
+	/* translators: 1: Item name, 2: Item type, 3: Item index, 4: Total Items */
+	'menuFocus'               => __( 'Edit %1$s (%2$s, %3$d of %4$d)' ),
+	/* translators: 1: Item name, 2: Item type, 3: Item index, 4: Total Items, 5: Item Parent. */
+	'subMenuFocus'            => __( 'Edit %1$s (%2$s, sub-item %3$d of %4$d of %5$s)' ),
```
Comment (joedolson) [inline src/wp-admin/nav-menus.php]: Change to `under %5$s`. Better English grammar, makes the string easier to parse while placeholders are present.

File: src/wp-admin/nav-menus.php [hunk 3]
```diff
@@ -580,10 +580,12 @@
...
+	/* translators: 1: Item name, 2: Item type, 3: Item index, 4: Total Items, 5: Item Parent, 6: Item Depth */
+	'subMenuMoreDepthFocus'   => __( 'Edit %1$s (%2$s, sub-item %3$d of %4$d of %5$s, level %6$d)' ),
```
Comment (joedolson) [inline src/wp-admin/nav-menus.php]: Change to `under %5$s`. Better English grammar, makes the string easier to parse while placeholders are present.

File: src/wp-includes/class-wp-customize-nav-menus.php [hunk 4]
```diff
@@ -562,10 +562,12 @@ public function enqueue_scripts() {
...
 			/* translators: %s: Previous item name. */
 			'outFrom'                 => __( 'Out from under %s' ),
-			/* translators: 1: Item name, 2: Item position, 3: Total number of items. */
-			'menuFocus'               => __( '%1$s. Menu item %2$d of %3$d.' ),
-			/* translators: 1: Item name, 2: Item position, 3: Parent item name. */
-			'subMenuFocus'            => __( '%1$s. Sub item number %2$d under %3$s.' ),
+			/* translators: 1: Item name, 2: Item type, 3: Item index, 4: Total Items */
+			'menuFocus'               => __( 'Edit %1$s (%2$s, %3$d of %4$d)' ),
+			/* translators: 1: Item name, 2: Item type, 3: Item index, 4: Total Items, 5: Item Parent. */
+			'subMenuFocus'            => __( 'Edit %1$s (%2$s, sub-item %3$d of %4$d of %5$s)' ),
```
Comment (joedolson) [inline src/wp-includes/class-wp-customize-nav-menus.php]: Change to `under %5$s`. Better English grammar, makes the string easier to parse while placeholders are present.

File: src/wp-includes/class-wp-customize-nav-menus.php [hunk 5]
```diff
@@ -562,10 +562,12 @@ public function enqueue_scripts() {
...
+			/* translators: 1: Item name, 2: Item type, 3: Item index, 4: Total Items, 5: Item Parent, 6: Item Depth */
+			'subMenuMoreDepthFocus'   => __( 'Edit %1$s (%2$s, sub-item %3$d of %4$d of %5$s, level %6$d)' ),
```
Comment (joedolson) [inline src/wp-includes/class-wp-customize-nav-menus.php]: Change to `under %5$s`. Better English grammar, makes the string easier to parse while placeholders are present.

File: src/wp-includes/customize/class-wp-customize-nav-menu-item-control.php [hunk 6]
```diff
@@ -77,10 +77,22 @@ public function content_template() {
...
-					?>
+					<# if ( 0 === data.depth ) { #>
+						<?php
+						/* translators: 1: Title of a menu item, 2: Type of a menu item */
+						printf( __( 'Edit %1$s (%2$s, %3$d of %4$d)' ), '{{ data.title || data.original_title || wp.customize.Menus.data.l10n.untitled }}', '{{ data.item_type_label }}', '', '' );
+						?>
+					<# } else if ( 1 === data.depth ) { #>
+						<?php
+							/* translators: 1: Title of a menu item, 2: Type of a menu item, 3: Depth of a menu item */
+							printf( __( 'Edit %1$s (%2$s, sub-item %3$d of %4$d of %5$s)' ), '{{ data.title || data.original_title || wp.customize.Menus.data.l10n.untitled }}', '{{ data.item_type_label }}', '', '', '' );
```
Comment (joedolson) [inline src/wp-includes/customize/class-wp-customize-nav-menu-item-control.php]: Change to `under %5$s`. Better English grammar, makes the string easier to parse while placeholders are present.

Also update the translator comment.

File: src/wp-includes/customize/class-wp-customize-nav-menu-item-control.php [hunk 7]
```diff
@@ -77,10 +77,22 @@ public function content_template() {
...
 				<span class="item-controls">
 					<button type="button" class="button-link item-edit" aria-expanded="false"><span class="screen-reader-text">
-					<?php
-						/* translators: 1: Title of a menu item, 2: Type of a menu item. */
-						printf( __( 'Edit menu item: %1$s (%2$s)' ), '{{ data.title || data.original_title || wp.customize.Menus.data.l10n.untitled }}', '{{ data.item_type_label }}' );
-					?>
+					<# if ( 0 === data.depth ) { #>
+						<?php
+						/* translators: 1: Title of a menu item, 2: Type of a menu item */
+						printf( __( 'Edit %1$s (%2$s, %3$d of %4$d)' ), '{{ data.title || data.original_title || wp.customize.Menus.data.l10n.untitled }}', '{{ data.item_type_label }}', '', '' );
```
Comment (joedolson) [inline src/wp-includes/customize/class-wp-customize-nav-menu-item-control.php:83]: Update the translator comment.

File: src/wp-includes/customize/class-wp-customize-nav-menu-item-control.php [hunk 8]
```diff
@@ -77,10 +77,22 @@ public function content_template() {
...
+						?>
+					<# } else if ( 1 === data.depth ) { #>
+						<?php
+							/* translators: 1: Title of a menu item, 2: Type of a menu item, 3: Depth of a menu item */
+							printf( __( 'Edit %1$s (%2$s, sub-item %3$d of %4$d of %5$s)' ), '{{ data.title || data.original_title || wp.customize.Menus.data.l10n.untitled }}', '{{ data.item_type_label }}', '', '', '' );
+						?>
+					<# } else { #>
+						<?php
+							/* translators: 1: Title of a menu item, 2: Type of a menu item, 3: Depth of a menu item */
+							printf( __( 'Edit %1$s (%2$s, sub-item %3$d of %4$d of %5$s, level %6$s)' ), '{{ data.title || data.original_title || wp.customize.Menus.data.l10n.untitled }}', '{{ data.item_type_label }}', '', '', '', '{{data.depth}}' );
```
Comment (joedolson) [inline src/wp-includes/customize/class-wp-customize-nav-menu-item-control.php]: Change to `under %5$s`. Better English grammar, makes the string easier to parse while placeholders are present.

Also update the translator comment.

File: src/js/_enqueues/wp/customize/nav-menus.js [hunk 9]
```diff
@@ -1574,6 +1576,84 @@
 			};
 		},
 
+		/**
+		 * 
+		 * Set up initial state of accessibility for menu items.
```
Comment (joedolson) [inline src/js/_enqueues/wp/customize/nav-menus.js]: Make this more explicit: it's setting up the initial state of the screen reader accessibility information for menu items. Setting up the state of accessibility implies it's doing things outside the parameters that it's actually handling.

**Review by joedolson — CHANGES_REQUESTED** [review]
This is in pretty good shape, and the remaining items are fairly small. The functionality works pretty well. It may still be possible to get it in, but it's getting pretty late. However, it seems like something that can definitely be ready for 6.7, if it doesn't make this release.

File: src/js/_enqueues/lib/nav-menu.js [hunk 10]
```diff
@@ -502,19 +503,24 @@
 			if ( isPrimaryMenuItem ) {
 				primaryItems = $( '.menu-item-depth-0' ),
 				itemPosition = primaryItems.index( menuItem ) + 1,
-				totalMenuItems = primaryItems.length,
+				totalMenuItems = primaryItems.length;
```
Comment (afercia) [inline src/js/_enqueues/lib/nav-menu.js]: Why this change? I think we should keep the comma? unless I'm missing something

File: src/js/_enqueues/lib/nav-menu.js [hunk 11]
```diff
@@ -502,19 +503,24 @@
...
 
 				// String together help text for primary menu items.
-				title = menus.menuFocus.replace( '%1$s', itemName ).replace( '%2$d', itemPosition ).replace( '%3$d', totalMenuItems );
+				title = menus.menuFocus.replace( '%1$s', itemName ).replace( '%2$s', menuItemType ).replace( '%3$d', itemPosition ).replace( '%4$d', totalMenuItems );
 			} else {
 				parentItem = menuItem.prevAll( '.menu-item-depth-' + parseInt( depth - 1, 10 ) ).first(),
 				parentItemId = parentItem.find( '.menu-item-data-db-id' ).val(),
 				parentItemName = parentItem.find( '.menu-item-title' ).text(),
 				subItems = $( '.menu-item .menu-item-data-parent-id[value="' + parentItemId + '"]' ),
+				totalSubItems = subItems.length;
```
Comment (afercia) [inline src/js/_enqueues/lib/nav-menu.js]: Why this line ends with a semicolon? I think we should use a comma? unless I'm missing something

File: src/js/_enqueues/lib/nav-menu.js [hunk 12]
```diff
@@ -502,19 +503,24 @@
...
 				itemPosition = $( subItems.parents('.menu-item').get().reverse() ).index( menuItem ) + 1;
 
 				// String together help text for sub menu items.
-				title = menus.subMenuFocus.replace( '%1$s', itemName ).replace( '%2$d', itemPosition ).replace( '%3$s', parentItemName );
+				if( depth < 2 ) {
```
Comment (afercia) [inline src/js/_enqueues/lib/nav-menu.js]: There should be a space after the `if`

File: src/js/_enqueues/lib/nav-menu.js [hunk 13]
```diff
@@ -732,6 +738,7 @@
 
 					api.refreshKeyboardAccessibility();
 					api.refreshAdvancedAccessibility();
+					api.refreshAdvancedAccessibilityOfItem( ui.item.find('a.item-edit') );
```
Comment (afercia) [inline src/js/_enqueues/lib/nav-menu.js]: There should be spaces inside the parenthesis: `( 'a.item-edit' )`

File: src/js/_enqueues/wp/customize/nav-menus.js [hunk 14]
```diff
@@ -1574,6 +1576,85 @@
 			};
 		},
 
+		/**
+		 *
```
Comment (afercia) [inline src/js/_enqueues/wp/customize/nav-menus.js]: This line is unnecessary.

File: src/js/_enqueues/wp/customize/nav-menus.js [hunk 15]
```diff
@@ -1574,6 +1576,85 @@
 			};
 		},
 
+		/**
+		 * 
+		 * It's setting up the initial state of the screen reader accessibility information for menu items.
```
Comment (afercia) [inline src/js/_enqueues/wp/customize/nav-menus.js]: It is preferred to start function descriptions with a third person verb e.g. 'Sets up ...'.

File: src/js/_enqueues/wp/customize/nav-menus.js [hunk 16]
```diff
@@ -1574,6 +1576,85 @@
...
+		 * Setting up the state of accessibility implies it's doing things outside the parameters that it's actually handling.
+		 * 
+		 * @since 6.6.0
+		 */
+		initAccessibility: function() {
+			var control = this,
+				menu = $( '#menu-to-edit' );
+
+			// Refresh the accessibility when the user comes close to the item in any way.
+			menu.on( 'mouseenter.refreshAccessibility focus.refreshAccessibility touchstart.refreshAccessibility' , '.menu-item' , function(){
```
Comment (afercia) [inline src/js/_enqueues/wp/customize/nav-menus.js]: Spaces before the commas should be removed. (I understand this code was copied from the existing implementation but yes the spaces should be removed)

File: src/js/_enqueues/wp/customize/nav-menus.js [hunk 17]
```diff
@@ -1574,6 +1576,85 @@
...
+				depth = menuItem.menuItemDepth(),
+				isPrimaryMenuItem = ( 0 === depth ),
+				itemName = $this.closest( '.menu-item-handle' ).find( '.menu-item-title' ).text(),
+				menuItemType = $this.closest( '.menu-item-handle' ).find( '.item-type' ).text(),
+				totalMenuItems = $('#menu-to-edit li').length;
+
+			if ( isPrimaryMenuItem ) {
+				primaryItems = $( '.menu-item-depth-0' ),
+				itemPosition = primaryItems.index( menuItem ) + 1,
+				totalMenuItems = primaryItems.length;
```
Comment (afercia) [inline src/js/_enqueues/wp/customize/nav-menus.js]: Same as above: why a semicolon?

File: src/js/_enqueues/wp/customize/nav-menus.js [hunk 18]
```diff
@@ -1574,6 +1576,85 @@
...
+
+				// String together help text for primary menu items.
+				title = menus.menuFocus.replace( '%1$s', itemName ).replace( '%2$s', menuItemType ).replace( '%3$d', itemPosition ).replace( '%4$d', totalMenuItems );
+			} else {
+				parentItem = menuItem.prevAll( '.menu-item-depth-' + parseInt( depth - 1, 10 ) ).first(),
+				parentItemId = parentItem.find( '.menu-item-data-db-id' ).val(),
+				parentItemName = parentItem.find( '.menu-item-title' ).text(),
+				subItems = $( '.menu-item .menu-item-data-parent-id[value="' + parentItemId + '"]' ),
+				totalSubItems = subItems.length;
```
Comment (afercia) [inline src/js/_enqueues/wp/customize/nav-menus.js]: Same as above: why a semicolon?

File: src/js/_enqueues/wp/customize/nav-menus.js [hunk 19]
```diff
@@ -1574,6 +1576,85 @@
...
+				itemPosition = $( subItems.parents('.menu-item').get().reverse() ).index( menuItem ) + 1;
+
+				// String together help text for sub menu items.
+				if( depth < 2 ) {
```
Comment (afercia) [inline src/js/_enqueues/wp/customize/nav-menus.js]: There should be a space after the `if`

File: src/js/_enqueues/wp/customize/nav-menus.js [hunk 20]
```diff
@@ -1574,6 +1576,85 @@
...
+					title = menus.subMenuFocus.replace( '%1$s', itemName ).replace( '%2$s', menuItemType ).replace( '%3$d', itemPosition ).replace( '%4$d', totalSubItems ).replace( '%5$s', parentItemName );
+				} else {
+					title = menus.subMenuMoreDepthFocus.replace( '%1$s', itemName ).replace( '%2$s', menuItemType ).replace( '%3$d', itemPosition ).replace( '%4$d', totalSubItems ).replace( '%5$s', parentItemName ).replace( '%6$d', depth );
+				}
+			}
+
+			$this.find('.screen-reader-text').text( title );
```
Comment (afercia) [inline src/js/_enqueues/wp/customize/nav-menus.js]: There should be spaces inside the parenthesis.

File: src/wp-admin/nav-menus.php [hunk 21]
```diff
@@ -580,10 +580,12 @@
 	'under'                   => __( 'Under %s' ),
 	/* translators: %s: Previous item name. */
 	'outFrom'                 => __( 'Out from under %s' ),
-	/* translators: 1: Item name, 2: Item position, 3: Total number of items. */
-	'menuFocus'               => __( '%1$s. Menu item %2$d of %3$d.' ),
-	/* translators: 1: Item name, 2: Item position, 3: Parent item name. */
-	'subMenuFocus'            => __( '%1$s. Sub item number %2$d under %3$s.' ),
+	/* translators: 1: Item name, 2: Item type, 3: Item index, 4: Total Items */
+	'menuFocus'               => __( 'Edit %1$s (%2$s, %3$d of %4$d)' ),
```
Comment (afercia) [inline src/wp-admin/nav-menus.php:584]: In `4: Total Item` the word Item should start with lowercase and end with a period.

File: src/wp-admin/nav-menus.php [hunk 22]
```diff
@@ -580,10 +580,12 @@
 	'under'                   => __( 'Under %s' ),
 	/* translators: %s: Previous item name. */
 	'outFrom'                 => __( 'Out from under %s' ),
-	/* translators: 1: Item name, 2: Item position, 3: Total number of items. */
-	'menuFocus'               => __( '%1$s. Menu item %2$d of %3$d.' ),
-	/* translators: 1: Item name, 2: Item position, 3: Parent item name. */
-	'subMenuFocus'            => __( '%1$s. Sub item number %2$d under %3$s.' ),
+	/* translators: 1: Item name, 2: Item type, 3: Item index, 4: Total Items */
```
Comment (afercia) [inline src/wp-admin/nav-menus.php]: In `4: Total Items` the word Items should start with lowercase and end with a period.

File: src/wp-admin/nav-menus.php [hunk 23]
```diff
@@ -580,10 +580,12 @@
...
+	'menuFocus'               => __( 'Edit %1$s (%2$s, %3$d of %4$d)' ),
+	/* translators: 1: Item name, 2: Item type, 3: Item index, 4: Total Items, 5: Item Parent. */
```
Comment (afercia) [inline src/wp-admin/nav-menus.php]: In `4: Total Items, 5: Item Parent`, Itmes and Parent should be lowercase.

File: src/wp-admin/nav-menus.php [hunk 24]
```diff
@@ -580,10 +580,12 @@
...
+	'subMenuFocus'            => __( 'Edit %1$s (%2$s, sub-item %3$d of %4$d under %5$s)' ),
+	/* translators: 1: Item name, 2: Item type, 3: Item index, 4: Total Items, 5: Item Parent, 6: Item Depth */
```
Comment (afercia) [inline src/wp-admin/nav-menus.php]: In `4: Total Items, 5: Item Parent, 6: Item Depth`, Items, Parend and Depth should be lowercase and hte comment should end with a period.

File: src/wp-includes/class-wp-customize-nav-menus.php [hunk 25]
```diff
@@ -562,10 +562,12 @@ public function enqueue_scripts() {
 			'under'                   => __( 'Under %s' ),
 			/* translators: %s: Previous item name. */
 			'outFrom'                 => __( 'Out from under %s' ),
-			/* translators: 1: Item name, 2: Item position, 3: Total number of items. */
-			'menuFocus'               => __( '%1$s. Menu item %2$d of %3$d.' ),
-			/* translators: 1: Item name, 2: Item position, 3: Parent item name. */
-			'subMenuFocus'            => __( '%1$s. Sub item number %2$d under %3$s.' ),
+			/* translators: 1: Item name, 2: Item type, 3: Item index, 4: Total Items */
+			'menuFocus'               => __( 'Edit %1$s (%2$s, %3$d of %4$d)' ),
+			/* translators: 1: Item name, 2: Item type, 3: Item index, 4: Total Items, 5: Item Parent. */
```
Comment (afercia) [inline src/wp-includes/class-wp-customize-nav-menus.php]: Same as above

File: src/wp-includes/class-wp-customize-nav-menus.php [hunk 26]
```diff
@@ -562,10 +562,12 @@ public function enqueue_scripts() {
...
+			'subMenuFocus'            => __( 'Edit %1$s (%2$s, sub-item %3$d of %4$d under %5$s)' ),
+			/* translators: 1: Item name, 2: Item type, 3: Item index, 4: Total Items, 5: Item Parent, 6: Item Depth */
```
Comment (afercia) [inline src/wp-includes/class-wp-customize-nav-menus.php]: Same as above

File: src/wp-includes/customize/class-wp-customize-nav-menu-item-control.php [hunk 27]
```diff
@@ -77,10 +77,22 @@ public function content_template() {
 				</span>
 				<span class="item-controls">
 					<button type="button" class="button-link item-edit" aria-expanded="false"><span class="screen-reader-text">
-					<?php
-						/* translators: 1: Title of a menu item, 2: Type of a menu item. */
-						printf( __( 'Edit menu item: %1$s (%2$s)' ), '{{ data.title || data.original_title || wp.customize.Menus.data.l10n.untitled }}', '{{ data.item_type_label }}' );
-					?>
+					<# if ( 0 === data.depth ) { #>
+						<?php
+						/* translators: 1: Title of a menu item, 2: Type of a menu item */
```
Comment (afercia) [inline src/wp-includes/customize/class-wp-customize-nav-menu-item-control.php]: Missing period at the end of translator comment

File: src/wp-includes/customize/class-wp-customize-nav-menu-item-control.php [hunk 28]
```diff
@@ -77,10 +77,22 @@ public function content_template() {
...
+						printf( __( 'Edit %1$s (%2$s, %3$d of %4$d)' ), '{{ data.title || data.original_title || wp.customize.Menus.data.l10n.untitled }}', '{{ data.item_type_label }}', '', '' );
+						?>
+					<# } else if ( 1 === data.depth ) { #>
+						<?php
+							/* translators: 1: Title of a menu item, 2: Type of a menu item, 3: Depth of a menu item */
```
Comment (afercia) [inline src/wp-includes/customize/class-wp-customize-nav-menu-item-control.php]: Missing period at the end of translator comment

**Comment by Rcreators** [comment]
@afercia All changes have been amended.

**Review by afercia — CHANGES_REQUESTED** [review]
Please see inline comments

**Comment by Rcreators** [comment]
> Please see inline comments

All the requested changes are pushed.

**Comment by afercia** [comment]
Thanks for addressing the coding standards points I raised @Rcreators 
I understand part of this PR is code copied from already existing code there's some more occurrences of missing spaces between parehthesis and single quote `('` or `')` that would be nice to fix if possible. Thanks!
Other than that, I would leave functional and final review to @joedolson

**Comment by joedolson** [comment]
In r58306

**Pull request closed without merging**